### PR TITLE
Update Jetpack Social media picker UI to match new designs

### PR DESCRIPTION
**Pull request by danielpost**
This PR implements the new design for the media picker UI in Jetpack Social.

## Proposed changes:

* Update media picker to match new design.
* Add option to remove picked media.
* Update error messages to be more descriptive.

### Other information:

- [X] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1HpG7-jqQ-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Add support for enhanced publishing to your site (context here: D90213-code)
* Start a new post.
* Make sure the design matches the screenshots.
* Check that you can pick an image.
* Check that you can replace an image by clicking on the current on.
* Check that you can remove an image.
* Make sure the tests pass: `cd projects/js-packages/publicize-components && npm test src/components/media-section/test/index.test.js`

image
image

### CODE REVIEW

File: projects/js-packages/publicize-components/src/components/media-section/index.js [hunk 1]
```diff
@@ -118,92 +136,96 @@ export default function MediaSection() {
...
+				return <div>Video Preview</div>;
+			}
+
+			const { width, height, sourceUrl } = mediaData;
+
+			if ( width && height && sourceUrl ) {
+				return (
+					<div className={ styles[ 'preview-wrapper' ] }>
+						<button className={ styles.remove } onClick={ onRemoveMedia }>
+							<VisuallyHidden>{ __( 'Remove media', 'jetpack' ) }</VisuallyHidden>
```
Comment (spsiddarthan) [inline projects/js-packages/publicize-components/src/components/media-section/index.js:152]: Is accessibility the purpose of the `VisuallyHidden` component?

File: projects/js-packages/publicize-components/src/components/media-section/index.js [hunk 2]
```diff
@@ -118,92 +136,96 @@ export default function MediaSection() {
...
-							<Text>{ ` ${ maxImageSize } Mb` }</Text>
-						</Notice>
-						<Text variant="title-small">{ __( 'Allowed types', 'jetpack' ) }</Text>
-						<Notice className={ styles.max_notice } isDismissible={ false } status="info">
-							<Text>{ ` ${ allowedMediaTypes.join( ', ' ) }` }</Text>
+		<ThemeProvider>
+			<BaseControl label={ __( 'Media', 'jetpack' ) } className={ styles.wrapper }>
+				<MediaUploadCheck>
+					<p className={ styles.subtitle }>
+						{ __( 'Choose a visual to accompany your post.', 'jetpack' ) }
```
Comment (spsiddarthan) [inline projects/js-packages/publicize-components/src/components/media-section/index.js:202]: Totally possible I am looking at an old iteration of the design, but I don't see the `Choose a visual to accompany your post.` in the design, do you think we should remove it?

File: projects/js-packages/publicize-components/src/components/media-section/index.js [hunk 3]
```diff
@@ -118,92 +136,96 @@ export default function MediaSection() {
...
+					</p>
+					<MediaUpload
+						title={ ADD_MEDIA_LABEL }
+						onSelect={ onUpdateMedia }
+						allowedTypes={ allowedMediaTypes }
+						render={ setMediaRender }
+						value={ attachedMedia[ 0 ]?.id }
+					/>
+					<ExternalLink href={ getRedirectUrl( 'jetpack-social-media-support-information' ) }>
```
Comment (spsiddarthan) [inline projects/js-packages/publicize-components/src/components/media-section/index.js:211]: The redirect url just takes me to MC now, do we plan to add it later?

**Review by spsiddarthan — COMMENTED** [review]
Nice work, @danielpost, tests per the instructions -  I didn't run the tests manually, but otherwise it tests well. 

I had some questions, but I am not 100% sure if they will have to be addressed in this PR or a different one.

- When you save the draft, and refresh the page, the image picker loses the selected media.
- When we republicize as well, we lose the already uploaded image/selected media.

File: projects/js-packages/publicize-components/src/components/media-section/index.js (hunk 1 above)
Comment (danielpost) [reply projects/js-packages/publicize-components/src/components/media-section/index.js:152]: Yup! Without it it's unclear what the button does for screen reader users.

File: projects/js-packages/publicize-components/src/components/media-section/index.js (hunk 2 above)
Comment (danielpost) [reply projects/js-packages/publicize-components/src/components/media-section/index.js:202]: It's missing from p1HpG7-jqQ-p2 but it was added in the wrap-up (see here: NIq2WbeH5khnwLVaHbUvSX-fi-3150%3A6735)

File: projects/js-packages/publicize-components/src/components/media-section/index.js (hunk 3 above)
Comment (danielpost) [reply projects/js-packages/publicize-components/src/components/media-section/index.js:211]: Yeah, that's a separate task: 1202411927350212-as-1203767773752680/f. Good catch!

**Comment by danielpost** [comment]
@spsiddarthan I can't reproduce the draft issue, do you have steps for me to reproduce? The image is still there after refreshing for me.

The Republicize one is interesting—I think we also discard the custom message so I'm wondering if we want to keep the image for a reshare or get rid of it. @pablinos what do you think?

**Comment by spsiddarthan** [comment]
It looks good except for the draft issue (which I have pinged about on the slack thread), @danielpost!

**Review by spsiddarthan — APPROVED** [review]
Paul pointed out that I was looking at the featured image when I was saving the draft and not attached social image, sorry about that! 

I think this is good to go!

File: projects/js-packages/publicize-components/src/components/media-section/index.js (hunk 3 above)
Comment (jeherve) [reply projects/js-packages/publicize-components/src/components/media-section/index.js:211]: I came here to say the same thing. Now that this has been merged, I would recommend enabling the redirect as quickly as possible.

File: projects/js-packages/publicize-components/src/components/media-section/test/index.test.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+				width: 300,
+				height: 700,
+			},
+		},
+	},
+};
+
+describe( 'MediaSection', () => {
+	it( 'should define the component', () => {
+		expect( 'MediaSection' ).toBeDefined();
```
Comment (anomiex) [inline projects/js-packages/publicize-components/src/components/media-section/test/index.test.js:87]: This does not test what it purports to test.

**Pull request merged**